ios-base/DroidKaigi 2020/Contributor/Views/ContributorViewDataSource.swift

```diff
@@ -0,0 +1,71 @@
+import ioscombined
+import Nuke
+import RxCocoa
+import RxSwift
+import UIKit
+
+final class ContributorViewDataSource: NSObject, UICollectionViewDataSource {
+    typealias Element = [ContributorIndex]
+
+    private var items: Element = []
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return items.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        switch kind {
+        case UICollectionView.elementKindSectionHeader:
+            guard let view = collectionView.dequeueReusableSupplementaryView(
+                ofKind: kind,
+                withReuseIdentifier: ContributorIndexHeaderView.identifier,
+                for: indexPath
+            ) as? ContributorIndexHeaderView else {
+                preconditionFailure()
+            }
+            view.indexLabel.text = items[indexPath.section].index.uppercased()
+            // for overlapping cells
+            view.frame.size = CGSize(width: UIScreen.main.bounds.width, height: ContributorCell.rowHeight)
+            view.clipsToBounds = false
+            return view
+
+        default:
+            preconditionFailure()
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return items[section].contributors.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ContributorCell.identifier, for: indexPath) as? ContributorCell else {
+            preconditionFailure()
+        }
+
+        let contributor = items[indexPath.section].contributors[indexPath.item]
+
+        if let url = URL(string: contributor.iconUrl) {
+            let options = ImageLoadingOptions(transition: .fadeIn(duration: 0.3))
+            Nuke.loadImage(with: url, options: options, into: cell.iconImageView)
+        }
+        cell.iconImageView.layer.cornerRadius = cell.iconImageView.frame.height / 2
```

Comment:
This is nits.
This code can write in `iconImageView`'s `didSet` in `ContributorCell ` 😄

Reply:
Right.